core/src/main/java/org/teavm/backend/wasm/generate/WasmDependencyListener.java

```diff
@@ -55,6 +68,20 @@ public void transformClass(ClassHolder cls, ClassReaderSource innerSource, Diagn
                 method.setProgram(null);
                 method.getModifiers().add(ElementModifier.NATIVE);
             }
+
+            AnnotationReader exportAnnot = method.getAnnotations().get(Export.class.getName());
+            if (exportAnnot != null) {
+                exportedMethods.add(method);
+
+            }
+        }
+    }
+
+    @Override
+    public void beforeCompleting(DependencyAgent agent) {
```

Comment:
You should be **very** careful when overriding `beforeCompleting`. This looks suspicious